lib/Doctrine/DBAL/Platforms/PostgreSqlPlatform.php

```diff
@@ -536,12 +537,26 @@ public function getAlterTableSQL(TableDiff $diff)
             if ($columnDiff->hasChanged('type') || $columnDiff->hasChanged('precision') || $columnDiff->hasChanged('scale') || $columnDiff->hasChanged('fixed')) {
                 $type = $column->getType();
 
+                // SERIAL/BIGSERIAL are not "real" types and we can't alter a column to that type
+                $columnDefinition = $column->toArray();
+                $columnDefinition['autoincrement'] = false;
+
                 // here was a server version check before, but DBAL API does not support this anymore.
-                $query = 'ALTER ' . $oldColumnName . ' TYPE ' . $type->getSQLDeclaration($column->toArray(), $this);
+                $query = 'ALTER ' . $oldColumnName . ' TYPE ' . $type->getSQLDeclaration($columnDefinition, $this);
                 $sql[] = 'ALTER TABLE ' . $diff->getName($this)->getQuotedName($this) . ' ' . $query;
             }
 
-            if ($columnDiff->hasChanged('default') || $columnDiff->hasChanged('type')) {
+            if ($columnDiff->fromColumn) {
```

Comment:
What do you think about extracting this `if` block to a private method? I think that something like would improve the readability a bit:

```php
private function hasDefaultTypeChanged(ColumnDiff $columnDiff, Column $newColumn) : bool
{
    $defaultTypeChanged = $columnDiff->hasChanged('type');

    if ( ! $columnDiff->fromColumn || ! $defaultTypeChanged) {
        return $defaultTypeChanged;
    }

    $oldTypeIsNumeric = $this->isNumericType($columnDiff->fromColumn->getType());
    $newTypeIsNumeric = $this->isNumericType($newColumn->getType());

    return ! ($oldTypeIsNumeric && $newTypeIsNumeric && $newColumn->getDefault() === null)
}
```

We might improve things there (specially naming wise), but I think it makes things easier and more understandable 😄